tally/include/tally/stopwatch.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+#pragma once
+
+#include <memory>
+
+#include "tally/stopwatch_recorder.h"
+
+namespace tally {
+
+class Stopwatch {
+ public:
+  Stopwatch(std::chrono::steady_clock::time_point,
+            std::shared_ptr<StopwatchRecorder> recorder);
+
+  // Stop stops the Stopwatch and records the duration of time it observed.
+  void Stop();
+
+ private:
+  const std::chrono::steady_clock::time_point start_;
```

Comment:
Avoid `const` field.

Reply:
Hmm, I think it's appropriate in this case since this a read-only field, no?

Reply:
As long as you don't assume people want to assign existing `Stopwatch` instances to other instances.